Master matteo

matteor6 opened the pull request:
- Vas-y wesh faut tester et merger
- Merci
- Tkt
- J'accepte volontiers une petite vidéo si le visuel du star crushing a wall marche bien :)

matteor6 committed:
Test Comment

matteor6 committed:
Refactoring NewLine and adding Star Crash visuals

- Created new global var NewLine (instead of local) => no need for pointers ;) ;)
- Created new function createNewLine that resets NewLine and might also fill it
- Generalized the star apparition (star only appears if gameStatus == 0)

- Moved 3 fixed variables of star mode to where star mode gets activated
- Some first efforts to generalize LEDMatrix size (some 9s and 10s replaced by variables)
- Some other typos corrected

matteor6 committed:
Flash walls when Star mode

- Flash walls White-Yellow when Star mode
- Changed car crash check to == Wall instead of != Black when Star mode on (if we add other stuff, we might only want to count the Walls)

matteor6 reviewed (commented):
Pushed a new commit with walls flashing when game is in star mode

matteor6 committed:
Updated comments

matteor6 committed:
Add possibiliby to display score in binary

Bugs:
- Probably debugged problem with newLine as one function's parameter -> it's now a global variable

New features:
- Create scoreDisplayMode variable to manage the score display mode
- If score display mode is in binary, the score will now show in binary in the top row of the screen

Optimizations:
- Moved gameStatus with other game variables

matteor6 commented:
Pushed a new commit with score display mode in binary + probably debug a problem with adding a new line. Please test and merge, bro

tcibils reviewed (commented):
- well played on setting the line creation as a function =D
 - show car crashing wall should be fine
 - flashing walls should work fine a priori, nice addition
 - score in binary shouldn’t have been in this branch but oh well, I understand you’re ambitious ^^ I’m too lazy to review the formulas though

Code review seems fine.
It still needs production testing though. Will do ASAP and provide video.

matteor6 committed:
Adding way to switch scoreDisplayMode while playing

- If up button is pressed, score display mode changes from decimal to binary or vice-versa
- I also had to add the upButton logic

- Moved reinitializations of leftButtonPushed and rightButtonPushed from the main loop to the function changeCarPosition

matteor6 commented:
- Loooool ouais j'aurais du faire une autre branche mais voilà quoi
- J'ai utilisé le up-button pour changer de scoreDisplayMode pendant le jeu!

matteor6 commented:
Toutes ces modifs et commits augmentent considérablement une chance d'erreur dans le code. Mais j'ai aussi retrouvé 1-2 potentiels bugs précédents en implémentant les nouvelles features. Donc peut-être que tout va bien ^^

matteor6 committed:
Optimization

- Will first check the game status, then create a random number. Might be a few nanoseconds faster or something.

matteor6 committed:
Updated to-do

- updated to-do

matteor6 committed:
Bugfix after testing

- Many empty spaces just after a star
- Less walls when star
- 10 points instead of 5 when star crushed a wall

matteor6 commented:
Bugfix after maman testing

matteor6 committed:
Optimization

- visual changes to the code, nothing functional

matteor6 committed:
Updated to-do

tcibils reviewed (commented):
- sympathique refacto de changeCarPosition avec les variables à 0
 - variable pour le nombre de colonnes vides, ce'st cool
 - utilisation du boutton "up" presque fonctionelle dans les tests ce matin : cliquer change bien de mode, mais il manque un reset je crois, certaines diodes restent allumées
 - clignotement des murs jaune/blanc si étoile pas fonctionnel, mais je sais pas si c'est souhaitable ? Peut être avec un jaune plus léger ?

(il me semblait bien qu'il y avait moins d'étoiles, filou !)

matteor6 committed:
Adding Secret feature & other fixes

- Activated down button
- Fixed "diode reste allumée quand on change de mode de binaire à linéaire"
- When star mode, walls are all shown in Yellow!!!
- Refactored numbers for gameStatuses (0 is game over, 1 is player select, 2 is normal, 3 is star)
- Updated to-do

Secret, don't read below!!!!!

- Created a gameStatus for player selecting mode
- Created letters M and L for you know who
- Created a Highlighting light for knowing which player is selected, and also alternating Red and Green highlight of letters M and L depending on who's selected
- Of course, Luigi is with his green car, it'sss meeee Luiiiigiii

matteor6 commented:
Fixes and stuff.
AND SECRET FEATURE.
Ne lis pas les commentaires si tu veux garder la surprise!!!

matteor6 committed:
Bugfixes

- Tweak star parameters (a bit more frequent, a bit less walls during it)
- Trying to fix initial mode

matteor6 committed:
Bugfixes

- Changed the 5-line player selection bar to a 3-line one
- Clears the matrix after player selection and before starting game
- Some comments

matteor6 commented:
Some bugfixes after testing

matteor6 committed:
Buxfixing final & updating to-do

- Probably fixed colors by re-ordering the White & Wall computations
- Showing the final score in White and Red again (hope it works)
- To-do updated

tcibils commented:
On merge sur master, ça te dis ?

matteor6 commented:
Ouaip! 👍

tcibils reviewed (approved):
Code working, nice modularization and added functionalities !
